connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -141,13 +142,36 @@ public void stop() {
     }
 
     // divide topic-partitions among tasks
+    // since each mirrored topic has different traffic and number of partitions, to balance the load
+    // across all mirrormaker instances (workers), 'roundrobin' helps to evenly assign all
+    // topic-partition to the tasks, then the tasks are further distributed to workers by calling
+    // 'ConnectorUtils.groupPartitions()'. For example, 3 tasks to mirror 3 topics with 8, 2 and 2
+    // partitions respectively. 't1' denotes 'task 1', 't0p5' denotes 'topic 0, partition 5'
+    // t1 -> [t0p0, t0p3, t0p6, t1p1]
+    // t2 -> [t0p1, t0p4, t0p7, t2p0]
+    // t3 -> [t0p2, t0p5, t1p0, t2p1]
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
         if (!config.enabled() || knownTopicPartitions.isEmpty()) {
             return Collections.emptyList();
         }
         int numTasks = Math.min(maxTasks, knownTopicPartitions.size());
-        return ConnectorUtils.groupPartitions(knownTopicPartitions, numTasks).stream()
+        List<List<TopicPartition>> roundRobinAssignment = new ArrayList<>(numTasks);
+        for (int i = 0; i < numTasks; i++) {
+            roundRobinAssignment.add(new ArrayList<>());
+        }
+        int count = 0;
+        for (TopicPartition partition : knownTopicPartitions) {
+            int index = count % numTasks;
+            roundRobinAssignment.get(index).add(partition);
+            count++;
+        }
+        List<TopicPartition> roundRobinFlatten = new ArrayList<>();
```

Comment:
isn't this a flatMap?

Reply:
I think this may not be a good variable, want to merge List<List<TopicParititon> into List<TopicPartition>,  let me update `roundRobinAssignment` to `roundRobinByTask`, `roundRobinFlatten` to something like "roundRobinAllTakss"?